parking_permits/views.py

```diff
@@ -394,13 +460,32 @@ def post(self, request, format=None):
             subscription.status = SubscriptionStatus.CANCELLED
             subscription.cancel_reason = request.data.get("reason")
             subscription.save()
-            for order in subscription.orders.all():
+            remaining_valid_subscriptions = subscription.order.subscriptions.filter(
+                status=SubscriptionStatus.CONFIRMED
+            )
+            if not remaining_valid_subscriptions.exists():
+                order = subscription.order
                 order.status = OrderStatus.CANCELLED
                 order.save()
-                permit_ids = order.permits.values_list("id", flat=True)
-                CustomerPermit(order.customer_id).end(
-                    permit_ids, ParkingPermitEndType.AFTER_CURRENT_PERIOD
+
+            permit = subscription.permit
+            # Create a refund for a remaining full month period, if it was charged already
+            if permit.end_time and permit.end_time - relativedelta(months=1) > tz.now():
```

Comment:
Should you add some kind of logging before this, to check why condition possible was false?

Reply:
Valid point, needs more logging.

Reply:
Added more logging for this.